_data-prepper/pipelines/configuration/processors/truncate.md

```diff
@@ -0,0 +1,107 @@
+
+## Configuration
+
+You can configure the `truncate` processor using the following options.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`entries` | Yes | String list | A list of entries to add to an event.
+`source_keys` | No | String list | The list of sources keys that will be modified by the processor. The default value is an empty list which indicates truncating all values.
+`truncate_when` | No | Conditional expression | A condition that, when met, determines when the truncate operation is performed. 
+`start_at` | No | Integer | Where inside the string value to start the truncation process. Default is `0`, which means the truncation of each key's value starts at the beginning.
+`length` | No | Integer| The length of the string after truncation. When not specified, the processor will measure the length based on where the string ends.
+
+For the `truncate` processor to run either the `start_at` or `length` options must be present in the configuration. For greater customization, you can define both values inside the configuration. 
+
+## Usage
+
+The following examples show how to configure the `truncate` processor inside the `pipeline.yaml` file:
+
+## Example: Minimum configuration
+
+The following example shows the minimum configuration of the `truncate` processor:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - truncate:
+        entries:
+          - source_keys: ["message1", "message2"]
+            length: 5
+          - source_keys: ["info"]
+            length: 6
+            start_at: 4
+          - source_keys: ["log"]
+            start_at: 5
+  sink:
+    - stdout:
+```
+
+For example, the following event contains several keys with string values:
+
+```json
+{"message1": "hello,world", "message2": "test message", "info", "new information", "log": "test log message"}
+```
+
+Then, the `truncate` processor produces the following output in which: 
+
+- The `message1` and `message 2` keys have the `start_at` setting set to `0`, indicating the truncation will being the start of the string, with the string itself truncated to a length of `5`.
+- The `info` key has the `start_at` setting set to `4`, indicating the truncation will begin at the `i` letter of the string, with the string truncated to a length of `6`.
+- The `log` key has a `start_at` setting set to `5` with no length indicated, meaning truncation will being at letter `l` of the string.
+
+```json
+{"message1":"hello", "message2":"test ", "info":"inform", "log": "log message"}
```

Comment:
Is this "the following output"? If so, it doesn't appear to show what we outline above.

Reply:
Ah OK, makes sense. I think I was confused by the example following so far after the reference, but it looks like it makes sense here.